tests/main/preseed-lxd/task.yaml

```diff
@@ -0,0 +1,97 @@
+summary: Check that preseeding of current ubuntu cloud image works under lxd.
+description: |
+  This test checks that preseeding of Ubuntu cloud images with snap-preseed
+  command works in lxc container. 
+
+# this test works only on 18.04 because it requires lxd from deb (lxd snap
+# as it wouldn't allow mount) and tries to replicate launchpad builder setup.
+systems: [ubuntu-18.04-*]
+
+environment:
+  IMAGE_MOUNTPOINT: /mnt/cloudimg
+
+prepare: |
+  IMGURL="https://storage.googleapis.com/spread-snapd-tests/images/cloudimg/focal-server-cloudimg-amd64.img"
+  wget --quiet "$IMGURL" -O cloudimg.img
+
+  modprobe nbd
+
+  # Run qemu-ndb as a service, so that it does not interact with ssh
+  # stdin/stdout it would otherwise inherit from the spread session.
+  systemd-run --system --service-type=forking --unit=qemu-ndb-preseed.service "$(command -v qemu-nbd)" --fork -c /dev/nbd0 "$(pwd)/cloudimg.img"
+  # nbd0p1 may take a short while to become available
+  retry-tool -n 5 --wait 1 test -e /dev/nbd0p1
+  mkdir -p "$IMAGE_MOUNTPOINT"
+  mount /dev/nbd0p1 "$IMAGE_MOUNTPOINT"
+
+  # Inject current snapd into seeds
+  #shellcheck source=tests/lib/snaps.sh
+  . "$TESTSLIB"/snaps.sh
+  SNAP_IMAGE=$(ls "$IMAGE_MOUNTPOINT"/var/lib/snapd/seed/snaps/snapd_*.snap)
+  unsquashfs "$SNAP_IMAGE"
+  cp /usr/lib/snapd/snapd squashfs-root/usr/lib/snapd/snapd
+  rm "$SNAP_IMAGE"
+  mksnap_fast squashfs-root "$SNAP_IMAGE"
+  sed -i "$IMAGE_MOUNTPOINT/var/lib/snapd/seed/seed.yaml" -E -e "s/^(\\s+)name: snapd/\\1name: snapd\\n\\1unasserted: true/"
+
+restore: |
+  lxc stop my-ubuntu --force
+  lxc delete my-ubuntu
+
+  umount "$IMAGE_MOUNTPOINT"
+  rmdir "$IMAGE_MOUNTPOINT"
+
+  # qemu-nbd -d may sporadically fail when removing the device,
+  # reporting it's still in use.
+  retry-tool -n 5 --wait 1 qemu-nbd -d /dev/nbd0
+
+  apt purge -y lxd
```

Comment:
It is important to check if we need to remove lxd before because if for some raeson it failed the prepare and lxd was not installed, then the restore will break discarding the current instance.  See next comment

Reply:
Done